### PR TITLE
feat: add option to use remote open api definition

### LINKED ISSUE #24: Flag to allow pulling openapi from the server dynamically rather than using the static version

### PR DESCRIPTION
**Pull request by mikewuu**
closes #24 

Add flag `--remote-api-defs` to pull the latest api defs from https://connect.getseam.com/openapi.json

To avoid fetching the remote definitions multiple times, we'll fetch it early on in the `cli` function:

```ts
const api = await getApiDefinitions(args.remote_api_defs)`
```

and pass it around everywhere it's needed:

```ts
 const selectedCommand = await interactForCommandSelection(api, args._)
```

### CODE REVIEW

File: cli.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { getSeam } from "./lib/get-seam"
 import chalk from "chalk"
 import { interactForServerSelection } from "./lib/interact-for-server-selection"
 import { getServer } from "./lib/get-server"
+import { getApiDefinitions } from "./lib/get-api-definitions"
```
Comment (seveibar) [inline cli.ts:14]: Seam projects use no semis and quotes, i think this got formatted, we should drop in a prettierrc

**Review by seveibar — COMMENTED** [review]
A bit hard to read because of the formatting change, we should drop a prettierrc

**Review by seveibar — APPROVED** [review]

File: cli.ts (hunk 1 above)
Comment (mikewuu) [reply cli.ts:14]: Yep, actually added prettier, but I'm hand-reverting the changes here to add it in a separate PR

**Comment by seveibar** [comment]
Some functions have over 2 params so we should switch to the "context helper" pattern imo with a context that contains the api, maybe the flags.


Eg interactForParams(cmd,ctx)

**Review by seveibar — APPROVED** [review]

**Pull request merged**